### PR TITLE
Improved agent Windows config. and secondary fixes

### PR DESCRIPTION
**Pull request by rshad**
Hi all!

This PR adds new improvements for the agent configuration in Windows and some fixes to the templates.

Originally some of these changes were added by @JPLachance in https://github.com/wazuh/wazuh-puppet/pull/189. But as we do not have write access to his fork branch  [coveord:feature/improved-agent-configuration](https://github.com/coveord/wazuh-puppet/tree/feature/improved-agent-configuration), then we ended up applying the merge with `wazuh:devel` branch in local and brought the source fork branch into a local branch in `wazuh:wazuh-puppet` called `feature-improved-agent-windows-config`.

#### Tasks

- [x] Merge wazuh:devel into the source branch of the user. 

- [x] Discard the unnecessary changes.  https://github.com/wazuh/wazuh-puppet/pull/189#issuecomment-577616495

- [x] Push changes to the user branch locally and create a new upstream branch as a copy of it. 

- [x] Adapt puppet manifests and templates to the default `ossec.conf` of `wazuh` `3.11.2`.

- [x] Verify the changes on 
- - [x] Windows
- - [x] Centos
- - [x] Ubuntu

Thanks, @JPLachance for contributing.

Kr,

Rshad

### CODE REVIEW

**Comment by rshad** [comment]
Hi all!

In this PR and as we mentioned before, we made the required changes to adapt the changes we picked up from the PR #189 and we also improved and parameterized almost all the configuration related to Wazuh agent for Windows OS. We also added some small improvements.

**Tasks**

- [ ] Parameterize the following templates related to `wazuh-agent` installation on `Windows`. 
- - `activeresponse`. _`new template`_
- - `rootcheck`.

- [ ] Adapt the following templates to be equal to the default config wazuh `3.11.2`.
- - `os-query`
- - `cis-cat` .
- - `syscollector`
- - `sca`
- - `agent.conf`
 
- [ ] Secondary changes and fixes for syntax errors `errors which were not detected before`.  

Kr,

Rshad

**Comment by rshad** [comment]
Hi all!

Here I attach the default 3.11.2 configuration files `ossec.conf` and wazuh-puppet `ossec.conf` for all of `CentOS`, `Ubuntu` and `Windows`.


`Note`: For some of the configuration files I reordered some config. parts so I could make the difference comparison correctly.

Kr,

Rshad
[config_files.zip](https://github.com/wazuh/wazuh-puppet/files/4109576/config_files.zip)

**Comment by rshad** [comment]
`ossec.conf` result files after the last changes.

[config_files.zip](https://github.com/wazuh/wazuh-puppet/files/4117273/config_files.zip)

File: manifests/params_agent.pp [hunk 1]
```diff
@@ -174,8 +182,10 @@
       $ossec_syscheck_auto_ignore = undef
       $ossec_syscheck_directories_1 = '/etc,/usr/bin,/usr/sbin'
       $ossec_syscheck_directories_2 = '/bin,/sbin,/boot'
-      $ossec_syscheck_whodata = '"no"'
-      $ossec_syscheck_realtime = '"no"'
+      $ossec_syscheck_whodata_directories_1 = '"no"'
```
Comment (jm404) [inline manifests/params_agent.pp]: Why double quoting? Please change it to `"no"` and fix recurrences

File: manifests/manager.pp [hunk 2]
```diff
@@ -569,7 +571,7 @@
     }
   }
 
-  if($ossec_syscheck_whodata == '"yes"') {
+  if ( $ossec_syscheck_whodata_directories_1 == '"yes"' ) or ( $ossec_syscheck_whodata_directories_2 == '"yes"' ) {
```
Comment (jm404) [inline manifests/manager.pp]: I don't get the double quoting, please review it

File: manifests/params_agent.pp (hunk 1 above)
Comment (rshad) [reply manifests/params_agent.pp]: Hi @jm404 !

Changes are applied and tested.

Kr,

Rshad

**Comment by rshad** [comment]
Hi all!

We also re-configured for both, the agent and the manager, the related configuration to the variables `whodata` and `realtime`, specifying a different variable of them for each of the default two lists of directories in `syscheck`. The template was changed so `realtime` and `whodata` will not appear in final `ossec.conf` if they got assigned the value `no`.


```ruby
      $ossec_syscheck_whodata_directories_1 = 'no'
      $ossec_syscheck_realtime_directories_1 = 'no'
      $ossec_syscheck_whodata_directories_2 = 'no'
      $ossec_syscheck_realtime_directories_2 = 'no'
```

https://github.com/wazuh/wazuh-puppet/blob/7264a35377c8a700411800545e846d40a0217bf7/templates/fragments/_syscheck.erb#L124-L129

Kr,

Rshad

**Comment by rshad** [comment]
Fixed Puppet linting errors and warnings.

image

**Pull request merged**